python/plugins/processing/tests/GdalAlgorithmsRasterTest.py

```diff
@@ -1659,6 +1659,20 @@ def testWarp(self):
                  source + ' ' +
                  outdir + '/check.jpg'])
 
+            # with target using custom projection and user-defined extent
+            custom_crs = 'proj4: +proj=utm +zone=36 +south +a=63785 +b=6357 +towgs84=-143,-90,-294,0,0,0,0 +units=m +no_defs'
+            self.assertEqual(
+                alg.getConsoleCommands({'INPUT': source,
+                                        'SOURCE_CRS': custom_crs,
+                                        'TARGET_CRS': custom_crs,
+                                        'TARGET_EXTENT': '18.67,18.70,45.78,45.81',
+                                        'TARGET_EXTENT_CRS': custom_crs,
+                                        'OUTPUT': outdir + '/check.jpg'}, context, feedback),
+                ['gdalwarp',
+                 '-s_srs "+proj=utm +zone=36 +south +a=63785 +b=6357 +towgs84=-143,-90,-294,0,0,0,0 +units=m +no_defs" -t_srs "+proj=utm +zone=36 +south +a=63785 +b=6357 +towgs84=-143,-90,-294,0,0,0,0 +units=m +no_defs" -r near -te 18.67 45.78 18.7 45.81 -te_srs "+proj=utm +zone=36 +south +a=63785 +b=6357 +towgs84=-143,-90,-294,0,0,0,0 +units=m +no_defs" -of JPEG ' +
+                 source + ' ' +
+                 outdir + '/check.jpg'])
+
```

Comment:
I would suggest you to create a distinct test. Basically copy this function, and remove elements that belong to the other test and validate elements related to your changes. This way we can more easily monitor when something breaks this functionality.